card/mod/bootstrap/set/abstract/bootswatch_theme.rb

```diff
@@ -0,0 +1,167 @@
+# Bootswatch themes are free themes for bootstrap available at https://bootswatch.com/.
+# For every bootswatch theme we have one card of card type "bootswatch skin".
+# They all have codenames following the pattern "#{theme_name}_skin".
+#
+# The original bootswatch theme is build from two files, `_variables.scss` and
+# `_bootswatch.scss`. The original bootstrap scss has to be put between those two.
+# `_variables.scss` overrides bootstrap variables, `_bootswatch.scss` overrides
+# bootstrap css (variables are defined with `!default` hence only the first appearance
+# has an effect, for css the last appearance counts)
+#
+# The content of a bootswatch theme card consists of four parts:
+#   * pre_variables: hard-coded theme independent stuff
+#       and bootstrap functions to make the available in the variables part
+#   * variables: the content from `_variables.scss`,
+#   * post_variables: the bootstrap css and libraries like select2 and
+#       bootstrap-colorpicker that depend on the theme
+#   * stylesheets: the content from `_bootswatch.scss` and custom styles
+#
+# For the original bootswatch themes all those parts are hard-coded and the content
+# is taken from files.
+# The bootswatch theme content is taken directly from the files in the bootswatch
+# submodule. For the rest we use code file cards.
+# Cards of type "customized skin" have the same structure but make the variables
+# and stylesheets part editable.
+#
+# Bootswatch theme cards are machine cards for the following reason.
+# Machine cards usually store all involved input cards of all nested levels in
+# there +*machine_input pointer. All those input cards
+# are processed separately and the result is joined to build the machine output.
+# That's a problem for this card when it's used as input.
+# A lot of the items depend on the variables scss and can't
+# be processed independently. Therefore we return only self as item card and join
+# the content of all the item cards in the content of the bootswatch theme card.
+# But then this card has to forward updates of its items to the machine cards it provides
+# input for.
+# To do that it is a machine itself and stores the generated machine output as its
+# content which will trigger the updates of other machines that use this card.
+
+include_set Abstract::Machine
+include_set Type::Scss
+include_set Abstract::CodeFile
+include_set Abstract::SkinThumbnail
+
+CONTENT_PARTS = %i[pre_variables variables post_variables stylesheets].freeze
```

Comment:
did you confirm that you're not concerned about conflicts with metrics in the meaning of "variables". this is a pretty narrow use of a very broad name...

Reply:
the name is "*variables", so it's not so broad, but wikirate's formula variables need some additional code to avoid conflicts. maybe "style: variables" is also an acceptable name